### PR TITLE
Allow to configure exclusion and inclusion patterns for rule sets

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/RuleSet.kt [hunk 1]
```diff
@@ -17,14 +20,24 @@ class RuleSet(val id: RuleSetId, val rules: List<BaseRule>) {
         validateIdentifier(id)
     }
 
+    var pathFilters: PathFilters? = null
+
     /**
      * Visits given file with all rules of this rule set, returning a list
      * of all code smell findings.
      */
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/RuleSet.kt]: Please update the comment to mention the path filters.

File: detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+                it("excludes file and does not report") {
+                    val config = TestConfig(Config.EXCLUDES_KEY to "**/*.kt")
+                    assertThat(ruleSetInstance(config).accept(file)).isEmpty()
+                }
+
+                it("should include and report file") {
+                    val config = TestConfig(
+                        Config.EXCLUDES_KEY to "**/*.kt",
+                        Config.INCLUDES_KEY to "**/*.kt"
```
Comment (schalkms) [inline detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt]: This confuses me:
`EXCLUDE` AND `INCLUDE` set to the same value

File: detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    )
+                    assertThat(ruleSetInstance(config).accept(file)).hasSize(1)
+                }
+            }
+        }
+    }
+})
+
+class TestProvider : RuleSetProvider {
```
Comment (schalkms) [inline detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt]: Maybe put this class into a separate file

File: detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    override val ruleSetId: String = "test"
+
+    override fun instance(config: Config): RuleSet = RuleSet(ruleSetId, listOf(Test()))
+}
+
+class Test : BaseRule() {
```
Comment (schalkms) [inline detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt]: Maybe put this class into a separate file

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/RuleSet.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/RuleSet.kt]: thx

File: detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt (hunk 2 above)
Comment (arturbosch) [reply detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt]: I provided a better test name for this.

File: detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt (hunk 3 above)
Comment (arturbosch) [reply detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/RuleSetSpec.kt]: I moved it inside the spec class and made it shorter.
No other test cases need the two classes so I prefer them inside this spec.

**Pull request merged**